step-functions-data-science-sdk/step_functions_mlworkflow_processing/step_functions_mlworkflow_scikit_learn_data_processing_and_model_evaluation.ipynb

```diff
@@ -0,0 +1,987 @@
+    "2. Run a training job using `TrainingStep` of AWS Step Functions Data Science SDK on the pre-processed training data to train a model\n",
+    "3. Run a processing job on the pre-processed test data to evaluate the trained model's performance using `ProcessingStep` of AWS Step Functions Data Science SDK\n",
+    "\n",
+    "The dataset used here is the [Census-Income KDD Dataset](https://archive.ics.uci.edu/ml/datasets/Census-Income+%28KDD%29). You select features from this dataset, clean the data, and turn the data into features that the training algorithm can use to train a binary classification model, and split the data into train and test sets. The task is to predict whether rows representing census responders have an income greater than `$50,000`, or less than `$50,000`. The dataset is heavily class imbalanced, with most records being labeled as earning less than `$50,000`. We train the model using logistic regression."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the latest sagemaker, stepfunctions and boto3 SDKs\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install --upgrade pip\n",
+    "!{sys.executable} -m pip install -qU awscli boto3 \"sagemaker>=1.71.0\"\n",
+    "!{sys.executable} -m pip install -qU \"stepfunctions>=1.1.0\"\n",
+    "!{sys.executable} -m pip show sagemaker stepfunctions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import the Required Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "from sagemaker.sklearn.processing import SKLearnProcessor\n",
+    "import time\n",
+    "import random\n",
+    "import uuid\n",
+    "import logging\n",
+    "import io\n",
+    "import random\n",
+    "import stepfunctions\n",
+    "import os\n",
+    "\n",
+    "from sagemaker.amazon.amazon_estimator import get_image_uri\n",
+    "from stepfunctions import steps\n",
+    "from stepfunctions.steps import TrainingStep, ModelStep, TransformStep, ProcessingStep\n",
+    "from stepfunctions.inputs import ExecutionInput\n",
+    "from stepfunctions.workflow import Workflow\n",
+    "from stepfunctions.template import TrainingPipeline\n",
+    "from stepfunctions.template.utils import replace_parameters_with_jsonpath\n",
+    "from sagemaker.processing import ProcessingInput, ProcessingOutput\n",
+    "from stepfunctions.steps import Chain\n",
+    "from sagemaker.s3 import S3Uploader\n",
+    "from stepfunctions.steps import ChoiceRule\n",
```

Comment:
`isort` should be applied here, as a best-practice: https://pypi.org/project/isort/

```suggestion
    "import io\n",
    "import logging\n",
    "import os\n",
    "import random\n",
    "import time\n",
    "import uuid\n",
    "\n",
    "import boto3\n",
    "import stepfunctions\n",
    "from stepfunctions import steps\n",
    "from stepfunctions.inputs import ExecutionInput\n",
    "from stepfunctions.steps import (\n",
    "    Chain,\n",
    "    ChoiceRule,\n",
    "    ModelStep,\n",
    "    ProcessingStep,\n",
    "    TrainingStep,\n",
    "    TransformStep,\n",
    ")\n",
    "from stepfunctions.template import TrainingPipeline\n",
    "from stepfunctions.template.utils import replace_parameters_with_jsonpath\n",
    "from stepfunctions.workflow import Workflow\n",
    "\n",
    "import sagemaker\n",
    "from sagemaker import get_execution_role\n",
    "from sagemaker.amazon.amazon_estimator import get_image_uri\n",
    "from sagemaker.processing import ProcessingInput, ProcessingOutput\n",
    "from sagemaker.s3 import S3Uploader\n",
    "from sagemaker.sklearn.processing import SKLearnProcessor\n",
```

Reply:
Will be happy to incorporate the changes and resolve them after the initial merge so that we can unblock the launch.